cobbler/api.py

```diff
@@ -930,7 +931,36 @@ def rename_menu(self, ref, newname: str):
 
     # ==========================================================================
 
-    # FIXME: add a new_item method
+    def new_item(self, what: str = "", is_subobject: bool = False):
+        """
+        Creates a new (unconfigured) object. The object is not persisted.
+
+        :param what: Specifies the type of object. Valid item types can be seen at
+                     :func:`~cobbler.enums.ItemTypes`.
+        :param is_subobject: If the object is a subobject of an already existing object or not.
+        :return: The newly created object.
+        """
+        if what == "distro":
+            new_item = self.new_distro(is_subobject=is_subobject)
+        elif what == "profile":
+            new_item = self.new_profile(is_subobject=is_subobject)
+        elif what == "system":
+            new_item = self.new_system(is_subobject=is_subobject)
+        elif what == "repo":
+            new_item = self.new_repo(is_subobject=is_subobject)
+        elif what == "image":
+            new_item = self.new_image(is_subobject=is_subobject)
+        elif what == "mgmtclass":
+            new_item = self.new_mgmtclass(is_subobject=is_subobject)
+        elif what == "package":
+            new_item = self.new_package(is_subobject=is_subobject)
+        elif what == "file":
+            new_item = self.new_file(is_subobject=is_subobject)
+        elif what == "menu":
+            new_item = self.new_menu(is_subobject=is_subobject)
+        else:
+            raise CX('internal error, collection name is "%s"' % what)
```

Comment:
What if try to do it this way? Just don't like repetative parts. Do you see any security concerns here?
```suggestion
        func = "new_{}".format(what)
        if what != "item" and hasattr(self, func):
            new_item = getattr(self, func)(is_subobject=is_subobject)
        else:
            raise CX('internal error, collection name is "%s"' % what)
```

Reply:
I agree with Victor, having all of these spelled out explicitly doesn't really make sense. As far as I can tell, there is one branch per `ItemType`. What do you think about:
```python
    try:
        ItemTypes(what) # verify that <what> is an ItemTypes member
        func = f"new_{what}"
        return getattr(self, func)(is_subobject=is_subobject)
    except (ValueError, AttributeError):
        raise Exception(f"internal error, collection name is {what}")
```
This has the upside that you don't need to add a new brach when you add a new member to `ItemTypes`